new/analyse/adapt-dependencies/spec/adapt-dependencies.spec.ts

```diff
@@ -0,0 +1,264 @@
+          ...envWithConfig,
+          packages: {
+            mockPackage1,
+            mockPackage2,
+            mockPackage3,
+            mockPackage4,
+          },
+        };
+
+        mockGetReleasablePackages.mockImplementationOnce(
+          () => mockReleasablePackages
+        );
+
+        mockSatisfies.mockImplementation(() => false);
+        mockMinVersion.mockImplementation(
+          () => ({ raw: '1.0.0' } as semver.SemVer)
+        );
+        mockInc.mockImplementation(() => '1.3.0');
+
+        result = adaptDependencies(mockEnv) as Env;
+      });
+
+      it('fetches releasable Packages', () => {
+        expect(mockGetReleasablePackages).toHaveBeenCalledTimes(1);
+        expect(mockGetReleasablePackages).toHaveBeenCalledWith(
+          mockEnv.packages
+        );
+      });
+
+      it('evaluates dependency ranges being satisfied', () => {
+        expect(mockSatisfies).toHaveBeenCalledTimes(1);
+        expect(mockSatisfies).toHaveBeenCalledWith('1.2.0', '>=1.0.0 <1.2.0');
+      });
+
+      it('adds a relatedMessage entry', () => {
+        expect(result.packages!.mockPackage4.relatedMessages[0]).toEqual({
+          date: new Date().toISOString(),
+          text: `fix: dependency "mockPackage2" has changed from 1.1.0 to 1.2.0`,
+          level: 'patch',
+        });
+      });
+
+      it('evaluates minVersion', () => {
+        expect(mockMinVersion).toHaveBeenCalledTimes(1);
+        expect(mockMinVersion).toHaveBeenCalledWith('>=1.0.0 <1.2.0');
+      });
+
+      it('evaluates upperVersionLimit', () => {
+        expect(mockInc).toHaveBeenCalledTimes(1);
+        expect(mockInc).toHaveBeenCalledWith('1.2.0', 'minor');
+      });
+
+      it('"dependingOnThis" dependency ranges are updated', () => {
+        expect(
+          result.packages!.mockPackage2.dependingOnThis[0].ownPackageRange
+        ).toEqual('>=1.0.0 <1.3.0');
+      });
+
+      it('"dependsOn" dependency ranges are updated', () => {
+        expect(result.packages!.mockPackage4.dependsOn[0].range).toEqual(
+          '>=1.0.0 <1.3.0'
+        );
+      });
+
+      it('returns an Env config object reflecting new ranges', () => {
+        const expectedEnv = clone(mockEnv);
+        expectedEnv.packages!.mockPackage2.dependingOnThis[0].ownPackageRange =
+          expectedEnv.packages!.mockPackage4.dependsOn[0].range =
+            '>=1.0.0 <1.3.0';
+        expectedEnv.packages!.mockPackage4.relatedMessages.push({
+          date: new Date().toISOString(),
+          text: `fix: dependency "mockPackage2" has changed from 1.1.0 to 1.2.0`,
+          level: 'patch',
+        });
+        expect(result).toEqual(expectedEnv);
+      });
+    });
+
+    // @TODO: Fill these cases
+    it.todo(
+      'throws exception if no package can be found on "dependsOn" collection'
+    );
+    it.todo('throws exception if lower version limit cannot be determined');
+    it.todo('throws exception if upper version limit cannot be determined');
```

Comment:
Upper limit is always calculated from the current version. So this will not be needed.